src/stan/io/array_var_context.hpp

```diff
@@ -268,13 +289,10 @@ class array_var_context : public var_context {
    * @param names Vector to store the list of names in.
    */
   virtual void names_r(std::vector<std::string>& names) const {
-    names.resize(0);
-    for (std::map<std::string, std::pair<std::vector<double>,
-                                         std::vector<size_t> > >::const_iterator
-             it
-         = vars_r_.begin();
-         it != vars_r_.end(); ++it)
-      names.push_back((*it).first);
+    names.clear();
```

Comment:
After the `clear()`, this should have `names.reserve(vars_r_.size())` so that the vector won't need to be resized as it's constructed.

This also applies to the next function.  So much so it might be nice to pull out the common functionality (but not necessary).